src/ASM/LR/Test/TestASMu2D.C

```diff
@@ -144,3 +145,58 @@ TEST(TestASMu2D, InterfaceChecker)
     }
   }
 }
+
+
+TEST(TestASMu2D, TransferGaussPtVars)
+{
+  SIM2D sim(1);
+  sim.opt.discretization = ASM::LRSpline;
+  sim.createDefaultModel();
+  ASMu2D* pch = static_cast<ASMu2D*>(sim.getPatch(1));
+  RealArray oldAr(9), newAr;
+  const double* xi = GaussQuadrature::getCoord(3);
+  size_t id[2];
+  for (size_t idx = 0; idx < 2; ++idx) {
+    SIM2D simNew(1);
+    simNew.opt.discretization = ASM::LRSpline;
+    simNew.createDefaultModel();
+    ASMu2D* pchNew = static_cast<ASMu2D*>(simNew.getPatch(1));
+    pchNew->uniformRefine(idx, 1);
+    for (id[1] = 0; id[1] < 3; ++id[1])
+      for (id[0] = 0; id[0] < 3; ++id[0])
+        oldAr[id[0]+id[1]*3] = (1.0 + xi[id[idx]]) / 2.0;
+    pchNew->transferGaussPtVars(pch->getSurface(), oldAr, newAr, 3);
+    size_t k = 0;
+    for (size_t iEl = 0; iEl < 2; ++iEl) {
+      for (id[1] = 0; id[1] < 3; ++id[1])
+        for (id[0] = 0; id[0] < 3; ++id[0], ++k)
+          EXPECT_FLOAT_EQ(newAr[k], 0.5*iEl + 0.5*(xi[id[idx]] + 1.0) / 2.0);
+    }
+  }
+}
+
+
+TEST(TestASMu2D, TransferGaussPtVarsN)
+{
+  SIM2D sim(1), sim2(1);
+  sim.opt.discretization = sim2.opt.discretization = ASM::LRSpline;
+  sim.createDefaultModel();
+  sim2.createDefaultModel();
+  ASMu2D* pch = static_cast<ASMu2D*>(sim.getPatch(1));
+  ASMu2D* pchNew = static_cast<ASMu2D*>(sim2.getPatch(1));
+  pchNew->uniformRefine(0, 1);
+  RealArray oldAr(9), newAr;
+  for (size_t i = 0; i < 9; ++i)
+    oldAr[i] = i + 1;
+
+  pchNew->transferGaussPtVarsN(pch->getSurface(), oldAr, newAr, 3);
+  static std::vector<double> refAr = {{1.0, 1.0, 2.0,
```

Comment:
Why not RealArray here also, since used elsewhere in this test (consistency)

Reply:
no reason, changed